sdk/src/main/java/io/dapr/client/DaprPreviewClient.java

```diff
@@ -224,4 +227,45 @@ <T> Mono<QueryStateResponse<T>> queryState(String storeName, Query query,
    * @return A Mono of QueryStateResponse of type T.
    */
   <T> Mono<QueryStateResponse<T>> queryState(QueryStateRequest request, TypeRef<T> type);
+
+  /**
+   * Publish multiple events to Dapr in a single request.
+   *
+   * @param request {@link BulkPublishRequest} object.
+   * @return A Mono of {@link BulkPublishResponse} object.
+   * @param <T> The type of events to publish in the call.
+   */
+  <T> Mono<BulkPublishResponse> publishEvents(BulkPublishRequest<T> request);
+
+  /**
+   * Publish multiple events to Dapr in a single request.
+   *
+   * @param pubsubName the pubsub name we will publish the event to.
+   * @param topicName the topicName where the event will be published.
+   * @param events the {@link List} of events to be published.
+   * @param contentType the content type of the event. Use Mime based types.
+   * @return the {@link BulkPublishResponse} containing publish status of each event.
+   *     The "entryID" field in {@link BulkPublishResponseEntry} in {@link BulkPublishResponse} will be
+   *     generated based on the order of events in the {@link List}.
+   * @param <T> The type of the events to publish in the call.
+   */
+  <T> Mono<BulkPublishResponse> publishEvents(String pubsubName, String topicName, List<T> events, String contentType);

```

Comment:
There may be a good reason why we do not already do this, but would it be possible to use `Optional` parameters instead of overriding methods and passing `null` values around?

Reply:
moved to using non null but default values ....